Use SnarkyJS version of 0.9.*

ymekuria opened the pull request:
Closes #374 

This was tested manually to confirm the correct SnarkyJS version in generated projects. It was also confirmed that the correct SnarkyJS version was displayed when running `zk system` .

mitschabaude reviewed (changes requested):
I think just using `0.9.*` doesn't achieve what we want. We want the CLI to install the latest snarkyjs version in new projects. It currently doesn't do that because `package-lock.json` locks the precise version.

So, what we need is to remove `template/project-ts/package-lock.json`. With that change, npm will determine the snarkyjs version to install from the version string in `package.json`.

The syntax of the version string is explained here: https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept

These are popular styles used:
* `^0.9.2` - what we have currently. npm will install any version that is newer than `0.9.2` but has the same major version `0`
* `~0.9.2` - npm will install any newer version that has the same _minor_ version `0.9`. 
  - documented alternatives without min patch version: `0.9`, `0.9.x`
  -  I assume `0.9.*` is the same as those, but not documented in the link above
  
If we want the second, `~` behavior, then I'd double-check the version string that you are using: the caret in `^0.9.*` suggests to me that any higher _minor_ version can be used. Also, the `.*` syntax is not documented on the npm page so I wouldn't use it.

I'd recommend to use `~0.9.2`, since we actually want a version higher than `0.9.2` (both `0.9.1` and `0.9.2` had important bug fixes). Alternatively, just `0.9` should achieve the same (because npm will pick the latest version that starts with `0.9`, so will be at least `0.9.2` anyway)

mitschabaude commented:
> I'd recommend to use ~0.9.2

Actually, I'm not sure why we'd want to prevent installing a higher minor version, so would slightly prefer `^0.9.2`, but curious what you @ymekuria or @jasongitmail think about that

jasongitmail reviewed (changes requested):
+1 the `package-lock.json` files will need to be untracked from git too. 

> Actually, I'm not sure why we'd want to prevent installing a higher minor version

@mitschabaude B/c we're in v0 and minor is _breaking_. Breaking upgrades should always be an intentional choice for devs and we should only update the patch version automatically for now.

> however, we didn't do that so far and no-one has complained, also it would be inconsistent with our current recommentation of running npm update snarkyjs in order to update to any newer version

Our written text to devs guiding during a breaking upgrade will just need to be updated. We're entering a new phase of maturity here network wise and SnarkyJS wise, and it's the right time to start providing more predictability around version upgrades for devs. 

`0.9.*` will work; confirmed with [NPM's version calculator](https://semver.npmjs.com/) (screenshot below). 
@ymekuria Let's remove `^` please, given the version calculator shows this works and it's the form that will be understood by the most devs, which is our goal here, agree?


<img width="721" alt="Screenshot 2023-03-08 at 7 29 00 PM" src="https://user-images.githubusercontent.com/50032291/223713762-83c38559-56c5-4318-9762-056e646005a8.png">

ymekuria commented:
> +1 the `package-lock.json` files will need to be untracked from git too.
@jasongitmail and @mitschabaude I updated the version to `0.9.*` and removed/untracked the `package-lock.json` files using `git rm package-lock.json`.  Now there are CI errors because running 'npm ci' [requires a package-lock.json](https://docs.npmjs.com/cli/v9/commands/npm-ci). I'm hesitant to use `npm install` in CI.

jasongitmail commented:
@ymekuria Good catch. DMing to talk through because there a bunch of aspects to consider here.
